File: cpp/Class.cpp
#include <iostream>
#include <string>

using namespace std;

class Film {
private:
    string id;
    string judul;
    string genre;
    string sutradara;
    int durasi; // menit
    int harga;  // rupiah

public:
    // Constructor default
    Film() {
        this->id = "";
        this->judul = "";
        this->genre = "";
        this->sutradara = "";
        this->durasi = 0;
        this->harga = 0;
    }

    // Constructor dengan parameter
    Film(string id, string judul, string genre, string sutradara, int durasi, int harga) {
        this->id = id;
        this->judul = judul;
        this->genre = genre;
        this->sutradara = sutradara;
        this->durasi = durasi;
        this->harga = harga;
    }

    // Getter & Setter
    void setId(string id) { this->id = id; }
    string getId() { return this->id; }

    void setJudul(string judul) { this->judul = judul; }
    string getJudul() { return this->judul; }

    void setGenre(string genre) { this->genre = genre; }
    string getGenre() { return this->genre; }

    void setSutradara(string sutradara) { this->sutradara = sutradara; }
    string getSutradara() { return this->sutradara; }

    void setDurasi(int durasi) { this->durasi = durasi; }
    int getDurasi() { return this->durasi; }

    void setHarga(int harga) { this->harga = harga; }
    int getHarga() { return this->harga; }

    // Method Tampilkan Data
    void tampilkanFilm() {
        cout << "ID Film   : " << id << endl;
        cout << "Judul     : " << judul << endl;
        cout << "Genre     : " << genre << endl;
        cout << "Sutradara : " << sutradara << endl;
        cout << "Durasi    : " << durasi << " menit" << endl;
        cout << "Harga     : Rp " << harga << endl;
        cout << "--------------------------------" << endl;
    }

    ~Film() {}
};

File: PHP/Class.php
<?php

class Film
{
	private string $id;
	private string $judul;
	private string $genre;
	private string $sutradara;
	private int $durasi;
	private int $harga;

	public function __construct(
		string $id = '',
		string $judul = '',
		string $genre = '',
		string $sutradara = '',
		int $durasi = 0,
		int $harga = 0
	) {
		$this->id = $id;
		$this->judul = $judul;
		$this->genre = $genre;
		$this->sutradara = $sutradara;
		$this->durasi = $durasi;
		$this->harga = $harga;
	}

	public function setId(string $id): void
	{
		$this->id = $id;
	}

	public function getId(): string
	{
		return $this->id;
	}

	public function setJudul(string $judul): void
	{
		$this->judul = $judul;
	}

	public function getJudul(): string
	{
		return $this->judul;
	}

	public function setGenre(string $genre): void
	{
		$this->genre = $genre;
	}

	public function getGenre(): string
	{
		return $this->genre;
	}

	public function setSutradara(string $sutradara): void
	{
		$this->sutradara = $sutradara;
	}

	public function getSutradara(): string
	{
		return $this->sutradara;
	}

	public function setDurasi(int $durasi): void
	{
		$this->durasi = $durasi;
	}

	public function getDurasi(): int
	{
		return $this->durasi;
	}

	public function setHarga(int $harga): void
	{
		$this->harga = $harga;
	}

	public function getHarga(): int
	{
		return $this->harga;
	}

	public function tampilkanFilm(): void
	{
		$barisBaru = PHP_SAPI === 'cli' ? PHP_EOL : '<br>';

		echo "ID Film   : {$this->id}{$barisBaru}";
		echo "Judul     : {$this->judul}{$barisBaru}";
		echo "Genre     : {$this->genre}{$barisBaru}";
		echo "Sutradara : {$this->sutradara}{$barisBaru}";
		echo "Durasi    : {$this->durasi} menit{$barisBaru}";
		echo "Harga     : Rp {$this->harga}{$barisBaru}";
		echo "--------------------------------{$barisBaru}";
	}
}